Tests/Marketplace/prepare_public_index_for_private_testing.py

```diff
@@ -105,13 +106,43 @@ def option_handler():
     parser.add_argument('-a', '--artifacts_path', help="The full path of packs artifacts", required=True)
     parser.add_argument('-ea', '--extract_artifacts_path', help="Full path of folder to extract wanted packs",
                         required=True)
-    parser.add_argument('-di', '--dummy_index_path', help="Full path to the dummy index in the private CI bucket",
+    parser.add_argument('-di', '--dummy_index_dir_path', help="Full path to the dummy index in the private CI bucket",
                         required=True)
 
     # disable-secrets-detection-end
     return parser.parse_args()
 
 
+def is_dummy_index_locked(public_storage_bucket, dummy_index_lock_path):
+    dummy_index_lock_blob = public_storage_bucket.blob(dummy_index_lock_path)
+    return dummy_index_lock_blob.exists()
+
+
+def lock_dummy_index(public_storage_bucket, dummy_index_lock_path):
+    dummy_index_lock_blob = public_storage_bucket.blob(dummy_index_lock_path)
+    with open('lock.txt', 'w') as lock_file:
+        lock_file.write('locked')
+
+    with open('lock.txt', 'rb') as lock_file:
+        dummy_index_lock_blob.upload_from_file(lock_file)
+
+
+def acquire_dummy_index_lock(public_storage_bucket, dummy_index_lock_path):
+    total_seconds_waited = 0
+    while is_dummy_index_locked(public_storage_bucket, dummy_index_lock_path):
+        if total_seconds_waited >= MAX_SECONDS_TO_WAIT_FOR_LOCK:
+            exit(1)
```

Comment:
https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python